readthedocs/projects/models.py

```diff
@@ -1570,6 +1570,43 @@ def save(self, *args, **kwargs):  # pylint: disable=arguments-differ
         super().save(*args, **kwargs)
 
 
+class HTTPHeader(TimeStampedModel, models.Model):
+
+    """
+    Define a HTTP header for a user Domain.
+
+    All the HTTPHeader(s) associated with the domain are added in the response
+    from El Proxito.
+
+    NOTE: the available headers are hardcoded in the NGINX configuration for
+    now (see ``dockerfile/nginx/proxito.conf``) until we figure it out a way to
+    expose them all without hardcoding them.
+    """
+
+    HEADERS_CHOICES = (
+        ('referrer_policy', 'Referrer-Policy'),
+        ('permissions_policy', 'Permissions-Policy'),
+        ('feature_policy', 'Feature-Policy'),
+        ('access_control_allow_origin', 'Access-Control-Allow-Origin'),
+        ('access_control_allow_headers', 'Access-Control-Allow-Headers'),
+        ('x_frame_options', 'X-Frame-Options'),
+    )
+
+    domain = models.ForeignKey(
+        Domain,
+        related_name='http_headers',
+        on_delete=models.CASCADE,
+    )
+    name = models.CharField(
+        max_length=128,
+        choices=HEADERS_CHOICES,
+    )
+    value = models.CharField(max_length=256)
+
```

Comment:
You could generically solve the `Strict-Transport-Security` only when secure by having a "only-set-if-secure" flag here if you want.

Reply:
I updated the PR adding a new field to handle this case.